bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/config/TestBigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,414 @@
+import static com.google.cloud.bigtable.config.RetryOptions.DEFAULT_BACKOFF_MULTIPLIER;
+import static com.google.cloud.bigtable.config.RetryOptions.DEFAULT_INITIAL_BACKOFF_MILLIS;
+import static com.google.cloud.bigtable.config.RetryOptions.DEFAULT_MAX_ELAPSED_BACKOFF_MILLIS;
+import static com.google.cloud.bigtable.config.RetryOptions.DEFAULT_MAX_SCAN_TIMEOUT_RETRIES;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnit4.class)
+public class TestBigtableVaneerSettingsFactory {
+
+  private static final Logger LOG = new Logger(TestBigtableVaneerSettingsFactory.class);
+
+  private static final String ACTUAL_PROJECT_ID = System.getProperty("test.client.project.id");
+  private static final String ACTUAL_INSTANCE_ID = System.getProperty("test.client.instance.id");
+
+  private static final String TEST_PROJECT_ID = "fakeProjectID";
+  private static final String TEST_INSTANCE_ID = "fakeInstanceID";
+  private static final String TEST_USER_AGENT = "sampleUserAgent";
+  private static final int SHORT_TIMEOUT_MS = 30_000;
+
+  /**
+   * RetryCodes for idempotent Rpcs.
+   */
+  private static final Set<Code> DEFAULT_RETRY_CODES =
+      ImmutableSet.of(Code.DEADLINE_EXCEEDED, Code.UNAVAILABLE);
+
+  private static final boolean endToEndArgMissing =
+      Strings.isNullOrEmpty(ACTUAL_PROJECT_ID) && Strings.isNullOrEmpty(ACTUAL_INSTANCE_ID);
+
+
+  @Rule
+  public ExpectedException expectException = ExpectedException.none();
+
+  private BigtableOptions bigtableOptions = BigtableOptions.builder()
+      .setProjectId(TEST_PROJECT_ID)
+      .setInstanceId(TEST_INSTANCE_ID)
+      .setUserAgent(TEST_USER_AGENT)
+      .setAdminHost("localhost")
+      .setDataHost("localhost")
+      .setCredentialOptions(CredentialOptions.nullCredential())
+      .setPort(8080)
+      .build();
+
+  private BigtableDataSettings dataSettings;
+  private BigtableTableAdminSettings adminSettings;
+  private BigtableDataClient dataClient;
+  private BigtableTableAdminClient adminClient;
+
+  @After
+  public void tearDown() throws Exception{
+    if (dataClient != null) {
+      dataClient.close();
+    }
+
+    if(adminClient != null){
+      adminClient.close();
+    }
+  }
+
+  public void initializeClients() throws IOException{
+    BigtableOptions bigtableOptions = BigtableOptions.builder()
+        .setProjectId(ACTUAL_PROJECT_ID)
+        .setInstanceId(ACTUAL_INSTANCE_ID)
+        .setUserAgent("native-bigtable-test-useragent")
+        .build();
+
+    dataSettings = BigtableVaneerSettingsFactory.createBigtableDataSettings(bigtableOptions);
+    dataClient = BigtableDataClient.create(dataSettings);
+
+    adminSettings = BigtableVaneerSettingsFactory.createTableAdminSettings(bigtableOptions);
+    adminClient = BigtableTableAdminClient.create(adminSettings);
+  }
+
+  /**
+   * This test will only run if it found "test.client.project.id" & "test.client.project.id"
```

Comment:
This test runs only if it finds "test.client.project.id" & "test.client.project.id"
   * VM arguments. Then it calls to an actual Bigtable Table & performs the checks below: